applications/tari_validator_node/src/main.rs

```diff
@@ -39,11 +46,14 @@ async fn main() {
 
     // Setup a panic hook which prints the default rust panic message but also exits the process. This makes a panic in
     // any thread "crash" the system instead of silently continuing.
-    let default_hook = panic::take_hook();
-    panic::set_hook(Box::new(move |info| {
-        default_hook(info);
-        process::exit(1);
-    }));
+    #[cfg(tari_feature_crash_on_thread_panic)]
+    {
+        let default_hook = panic::take_hook();
+        panic::set_hook(Box::new(move |info| {
+            default_hook(info);
+            process::exit(1);
+        }));
+    }
```

Comment:
The use here of `#[cfg(tari_feature_crash_on_thread_panic)]` instead of `#[tari_feature(crash_on_thread_panic)]` is because I could not successfully re-write the macro to manage syn::Item's as well as expressions. It may yet be possible but I thought it more important to push the PR than to spend more time debugging that detail.